homeassistant/components/bang_olufsen/websocket.py

```diff
@@ -101,6 +109,28 @@ def on_active_listening_mode(self, notification: ListeningModeProps) -> None:
             notification,
         )
 
+    def on_beo_remote_button_notification(self, notification: BeoRemoteButton) -> None:
+        """Send beo_remote_button dispatch."""
+        # Trigger the device trigger
+        self.hass.bus.async_fire(
+            BANG_OLUFSEN_EVENT,
+            event_data={
+                CONF_TYPE: f"{notification.key}_{notification.type}",
+                CONF_DEVICE_ID: self._device.id,
+            },
+        )
+
+    def on_button_notification(self, notification: ButtonEvent) -> None:
+        """Send button dispatch."""
+        # Trigger the device trigger
+        self.hass.bus.async_fire(
+            BANG_OLUFSEN_EVENT,
+            event_data={
+                CONF_TYPE: f"{notification.button}_{notification.state}",
+                CONF_DEVICE_ID: self._device.id,
+            },
+        )
```

Comment:
I'd suggest to, in an initial PR, add [event entities](https://developers.home-assistant.io/docs/core/entity/event), where each button is one event entity which can fire different events.

Also, should the beo remotes be their own devices?

Reply:
I guess I could add the Beoremote One as a device, I'll take a look at that. I didn't even know event Entities were a thing, so I'll take a look at that as well. I'll convert this PR to a draft for now.

Reply:
OK, I've pushed an initial commit containing event entities for the physical buttons as well as Beoremote One keys and a device for the Beoremote One. There are 10 Event entities for the physical buttons and 58 for Beoremote One keys. This seems to be a bit excessive, but still seems to be a better way to handle events. What are your thoughts?

Reply:
I agree it's not great. Maybe to add tens of entities which are not very likely to be used. Maybe make the event entities disabled, then users who want to use them can enable them?